frontend/app-development/hooks/useOpenSettingsModalBasedQueryParam.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useSettingsModalContext } from '../contexts/SettingsModalContext';
+import type { SettingsModalTabId } from '../types/SettingsModalTabId';
+import { allSettingsModalTabs } from '../layout/PageHeader/SubHeader/SettingsModalButton/SettingsModal/hooks/useSettingsModalMenuTabConfigs';
+
+export const queryParamKey: string = 'openSettingsModalWithTab';
+
+export function useOpenSettingsModalBasedQueryParam(): void {
+  const [searchParams] = useSearchParams();
+  const { settingsRef } = useSettingsModalContext();
+
+  useEffect((): void => {
+    const tabToOpen: SettingsModalTabId = searchParams.get(queryParamKey) as SettingsModalTabId;
+    const shouldOpenModal: boolean = isValidTab(tabToOpen);
+    if (shouldOpenModal) {
+      settingsRef.current.openSettings(tabToOpen);
+    }
+  }, [searchParams]);
```

Comment:
Should we look into this comment from Github Actions? 😄 
And if it should not be added, maybe we can add a comment explaining why?

Reply:
I added it to make sure the linter i happy, since it's a ref, it would not re-trigger the use effect, but it's not "wrong" to include it in the dependency array. 😊

Reply:
A happy linter is nice 😄